checker/dns/cname/cname.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 The Wait4X Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cname
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"regexp"
+
+	"wait4x.dev/v2/checker"
+)
+
+// Option configures an DNS CNAME record
+type Option func(d *CNAME)
+
+// CNAME represents DNS CNAME data structure
+type CNAME struct {
+	nameserver      string
+	address         string
+	expectedDomains []string
+	resolver        *net.Resolver
+}
+
+// New creates the DNS CNAME checker
+func New(address string, opts ...Option) checker.Checker {
+	d := &CNAME{
+		address: address,
+	}
+
+	// apply the list of options to CNAME
+	for _, opt := range opts {
+		opt(d)
+	}
+
+	// Nameserver settings.
+	d.resolver = net.DefaultResolver
```

Comment:
I believe you can specify the default resolver during struct initialization.
```suggestion
	d := &CNAME{
		address: address,
		resolver: net.DefaultResolver,
	}

	// apply the list of options to CNAME
	for _, opt := range opts {
		opt(d)
	}
```